cli/src/commands/git/fetch.rs

```diff
@@ -98,6 +100,15 @@ pub fn cmd_git_fetch(
             GitFetchError::InternalGitError(err) => map_git_error(err),
             _ => user_error(err),
         })?;
+        for pattern in &args.branch {
```

Comment:
Open question: This will issue a warning for *each* remote. So if a user has multiple remotes configured in `git.fetch` in the config file, this will issue a warning for a remote where the branch was not found, even if it finds it for the other remote(s).

thoughts?

Reply:
I would have thought that being in one remote would be enough to not get a warning, as it means this is probably not a typo.

Reply:
I'm not sure I understand your answer @PhilipMetzger. If `git.fetch = ["origin", "upstream"]` and the user issues `git fetch -b my-topic-branch`, do you think it warrants a warning that `my-topic-branch` is not found on upstream, even if it is found in `origin`? IMO the intention of the user is clear here.

Reply:
> IMO the intention of the user is clear here.

I'm not so sure it's that clear. I think any of the below could be true:

* fetch `my-topic-branch` from only `origin`?
* fetch `my-topic-branch` from both `origin` and `upstream`, so they can diff/cherrypick locally?
* actually meant to fetch from `upstream` but didn't realize it existed on `origin`. Whoops!

Life would be much simpler if remote branches were always specified as <branch>@<remote>, alas...

Reply:
> being in one remote would be enough to not get a warning, as it means this is probably not a typo.

+1

> if a branch glob doesn't match any of the changed or known remote branches

I just said that the implementation would have to scan changed refs as well as the known `repo.view()` (or `old_repo.view()` + `new_repo.view()` might be simpler.)

Reply:
This is a matter of taste I guess, but I usually display warnings only when both conditions are true:
- There is a high probability that the user did something wrong (otherwise, there would be too many false positive, and the warning becomes annoying) – for example using `jj log XXX` where `XXX` doesn't match any file, and is probably meant to be a revset.
- I cannot prove that they did something wrong (otherwise, that would be an error, not a warning) – the user might have really wanted to check if any revision had touched file `XXX` in `jj log XXX`.

If the glob expands to at least one branch on one remote, I'm afraid I couldn't determine with a high probability that it is indeed a mistake, as such use would be valid. I, for one, would not warn in this case.

Reply:
> Thanks for laying out your thoughts. This is convincing to me. Especially as you and @yuja think the same. I'll go ahead and change the warning to only fail if missing from _all_ remotes.

Done.

> I just said that the implementation would have to scan changed refs as well as the known repo.view()

Switched the implementation to this. Thanks for the hint. So there are no library changes now, only cli changes.